core/search/nested_filter.cpp

```diff
@@ -425,10 +470,27 @@ class MinMatcher : public Merger,
   doc_id_t Accept(const doc_id_t first_child, const doc_id_t parent) {
     assert(!doc_limits::eof(parent));
 
+    if (0 == min_) {
+      if constexpr (HasScore_v<Merger>) {
+        // Reset score value as we might not be able
+        // to find any childs
+        auto& merger = static_cast<Merger&>(*this);
+        auto& buf = static_cast<BufferType&>(*this);
+        std::memset(buf.data(), 0, merger.byte_size());
+      }
+      return 0;
+    }
+
     if (first_child > parent) {
       return first_child;
     }
 
+    doc_id_t count = min_ - 1;
+
+    if (!count) {
```

Comment:
Can you explain to me why we cannot check it before `first_child > parent` check?